CvGameCoreDLL_Expansion2/CvGame.cpp

```diff
@@ -8354,9 +8368,23 @@ void CvGame::updateMoves()
 		if (isOption(GAMEOPTION_DYNAMIC_TURNS) || isOption(GAMEOPTION_SIMULTANEOUS_TURNS))
 		{//Activate human players who are playing simultaneous turns now that we've finished moves for the AI.
 			// KWG: This code should go into CheckPlayerTurnDeactivate
+			size_t Oi;
+			int processOrder[MAX_PLAYERS];
+			for (Oi = 0; Oi < MAX_PLAYERS; Oi++) 
+			{
+				processOrder[Oi] = Oi;
+			}
+			for (Oi = 0; Oi < MAX_PLAYERS - 1; Oi++) 
```

Comment:
I'd yank this into a shuffle method. Personally I'd implement it just like https://en.wikipedia.org/wiki/Fisher%E2%80%93Yates_shuffle#The_modern_algorithm and comment about it being a Fisher-Yates shuffle. Shuffling is notoriously easy to screw up.